spec/acceptance/mod_suphp_spec.rb

```diff
@@ -1,6 +1,19 @@
 require 'spec_helper_acceptance'
 
+# why is that not executed on 16.04
 describe 'apache::mod::suphp class', if: (fact('operatingsystem') == 'Ubuntu' && fact('operatingsystemmajrelease') != '16.04') do
+  if ['16.04', '18.04'].include?(fact('operatingsystemmajrelease'))
+    # this policy defaults to 101. it prevents newly installed services from starting
+    # it is useful for containers, it prevents new processes during 'docker build'
+    # but we actually want to test the services and this should not behave like docker
+    # but like a normal operating system
+
+    # without this apache fails to start -> installation of mod-php-something fails because it reloads apache to enable the module. WTF Ubuntu
+    # exit codes are documented at https://askubuntu.com/a/365912. Default for docker images is 101
+    shell("sed -i 's/^exit.*/exit 0/' /usr/sbin/policy-rc.d")
```

Comment:
Would it be worth adding this code into the spec_helper_acceptance since you reuse it a couple of times? Also gross to these bugs :|  I am running this PR through an ad-hoc pipeline and am hitting this issue on Ubuntu 16 as you seem to be.

Reply:
Did you run the master branch and that failed, or my version?

Reply:
Your version. I'll re-run again this morning with your recent changes :)

Reply:
So I've found locally that my tests will run if I change to 
`  before :all do
    on(host, "if [ -a '/usr/sbin/policy-rc.d' ]; then sed -i 's/^exit.*/exit 0/' /usr/sbin/policy-rc.d; fi") if ['16.04', '18.04'].include?(fact('operatingsystemmajrelease'))
  end`
instead of doing it in the describe.

Reply:
Ew that didn't format well.. To summarise I just moved it into a 'before :all' clause. That seems to be why I kept hitting into that issue where the tests bombed out instead of running.

Reply:
I added it like this:

```patch
 spec/acceptance/mod_suphp_spec.rb | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/spec/acceptance/mod_suphp_spec.rb b/spec/acceptance/mod_suphp_spec.rb
index dbf3a28c..4a1388b8 100644
--- a/spec/acceptance/mod_suphp_spec.rb
+++ b/spec/acceptance/mod_suphp_spec.rb
@@ -1,8 +1,8 @@
 require 'spec_helper_acceptance'
 
 # why is that not executed on 16.04
-describe 'apache::mod::suphp class', if: (fact('operatingsystem') == 'Ubuntu' && fact('operatingsystemmajrelease') != '16.04') do
-  if ['16.04', '18.04'].include?(fact('operatingsystemmajrelease'))
+describe 'apache::mod::suphp class' do
+  before :all do
     # this policy defaults to 101. it prevents newly installed services from starting
     # it is useful for containers, it prevents new processes during 'docker build'
     # but we actually want to test the services and this should not behave like docker
@@ -10,8 +10,9 @@ describe 'apache::mod::suphp class', if: (fact('operatingsystem') == 'Ubuntu' &&
 
     # without this apache fails to start -> installation of mod-php-something fails because it reloads apache to enable the module. WTF Ubuntu
     # exit codes are documented at https://askubuntu.com/a/365912. Default for docker images is 101
-    shell("sed -i 's/^exit.*/exit 0/' /usr/sbin/policy-rc.d")
+    on(host, "if [ -a '/usr/sbin/policy-rc.d' ]; then sed -i 's/^exit.*/exit 0/' /usr/sbin/policy-rc.d; fi") if ['16.04', '18.04'].include?(fact('operatingsystemmajrelease'))
   end
+
   # mod php isn't supported for ubuntu 1.04
   next if fact('operatingsystemmajrelease') == '18.04'
   context 'default suphp config' do
```

but I get this error:

```
Failures:

  1) apache::mod::suphp class default suphp config succeeds in puppeting suphp
     Failure/Error: on(host, "if [ -a '/usr/sbin/policy-rc.d' ]; then sed -i 's/^exit.*/exit 0/' /usr/sbin/policy-rc.d; fi") if ['16.04', '18.04'].include?(fact('operatingsystemmajrelease'))
     NoMethodError:
       private method `exec' called for Host "":Serverspec::Type::Host
```